src/plugins/python/pythonplugin.cpp

```diff
@@ -102,35 +104,20 @@ void PythonPlugin::initialize()
             return;
         }
 
-        // w/o differentiating error messages could just rename "log"
-        // to "write" in the binding and assign plugin directly to stdout/stderr
-        PySys_SetObject((char *)"_tiledplugin",
+        // this is written to by stdout/stderr redirection in _preload.py
+        PySys_SetObject((char *)"_logger",
                         _wrap_convert_c2py__Tiled__LoggingInterface(&mLogger));
-
-        PyRun_SimpleString("import sys\n"
-                           "#from tiled.Tiled.LoggingInterface import INFO,ERROR\n"
-                           "class _Catcher:\n"
-                           "   def __init__(self, type):\n"
-                           "      self.buffer = ''\n"
-                           "      self.type = type\n"
-                           "   def write(self, msg):\n"
-                           "      self.buffer += msg\n"
-                           "      if self.buffer.endswith('\\n'):\n"
-                           "         sys._tiledplugin.log(self.type, self.buffer)\n"
-                           "         self.buffer = ''\n"
-                           "sys.stdout = _Catcher(0)\n"
-                           "sys.stderr = _Catcher(1)\n");
-
-        PyRun_SimpleString(QString("import sys; sys.path.insert(0, \"%1\")")
-                           .arg(mScriptDir).toUtf8().constData());
-
-        log(QString("-- Added %1 to path\n").arg(mScriptDir));
     }
 
-    reloadModules();
+    reloadModules(mSysScriptDir);
+    reloadModules(mUserScriptDir);
+
+    if (QFile::exists(mUserScriptDir)) {
+        mFileSystemWatcher.addPath(mUserScriptDir);
 
-    if (QFile::exists(mScriptDir))
-        mFileSystemWatcher.addPath(mScriptDir);
+        connect(&mFileSystemWatcher, SIGNAL(directoryChanged(QString)),
+                &mReloadTimer, SLOT(start()));
```

Comment:
This connection is now already made in the constructor, so it can be removed here.